ReactAndroid/src/main/java/com/facebook/react/uimanager/animation/FrameBasedAnimationDriver.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.uimanager.animation;
+
+import com.facebook.react.bridge.ReadableArray;
+import com.facebook.react.bridge.ReadableMap;
+
+/**
+ * Implementation of {@link AnimationDriver} which provides a support for simple time-based
+ * animations that are pre-calculate on the JS side. For each animation frame JS provides a value
+ * from 0 to 1 that indicates a progress of the animation at that frame.
+ */
+class FrameBasedAnimationDriver extends AnimationDriver {
+
+  private long mStartFrameTimeNanos = -1;
+  private final double[] mFrames;
+  private final double mToValue;
+  private double mFromValue;
+  private boolean mHasToValue;
+
+  FrameBasedAnimationDriver(ReadableMap config) {
+    ReadableArray frames = config.getArray("frames");
+    int numberOfFrames = frames.size();
+    mFrames = new double[numberOfFrames];
+    for (int i = 0; i < numberOfFrames; i++) {
+      mFrames[i] = frames.getDouble(i);
+    }
+    if (config.hasKey("toValue")) {
+      mHasToValue = true;
+      mToValue = config.getDouble("toValue");
+    } else {
+      mHasToValue = false;
+      mToValue = Double.NaN;
+    }
+  }
+
+  public boolean runAnimationStep(long frameTimeNanos) {
+    if (mStartFrameTimeNanos < 0) {
+      mStartFrameTimeNanos = frameTimeNanos;
+      mFromValue = mAnimatedValue.mValue;
+    }
+    long timeFromStartNanos = (frameTimeNanos - mStartFrameTimeNanos);
+    // frames are calculated at 60FPS, to get index by a given time offset from the start of the
+    // animation, we take the time diff in millisecond and divide it by 60 frames per 1000ms.
+    int frameIndex = (int) (timeFromStartNanos / 1000000L * 60L / 1000L);
```

Comment:
This probably doesn't matter for this diff since we probably won't have animations that run long enough, but this isn't necessarily safe. Phones run at **about** 60FPS, but can vary (e.g. I think my Nexus 5 runs at ~59FPS optimally). For our internal framerate logging, we generally keep track of when we saw the last frame and use that to try to calculate how many frames occurred since then by dividing by the estimated frame time (1000 / 60ms). This will generally keep you from accumulating errors over time that you get by assuming exactly 60FPS.